llvm/tools/llvm-exegesis/lib/RISCV/Target.cpp

```diff
@@ -156,15 +156,33 @@ template <class BaseT> class RISCVSnippetGenerator : public BaseT {
     // FIXME: We could have obtained these two constants from RISCVSubtarget
     // but in order to get that from TargetMachine, we need a Function.
     const MCSubtargetInfo &STI = State.getSubtargetInfo();
-    ELEN = STI.checkFeatures("+zve64x") ? 64 : 32;
-
-    std::string ZvlQuery;
-    for (unsigned Size = 32; Size <= 65536; Size *= 2) {
-      ZvlQuery = "+zvl";
-      raw_string_ostream SS(ZvlQuery);
-      SS << Size << "b";
-      if (STI.checkFeatures(SS.str()) && ZvlVLen < Size)
-        ZvlVLen = Size;
+    ELEN = STI.hasFeature(RISCV::FeatureStdExtZve64x) ? 64 : 32;
+
+    static_assert(RISCV::FeatureStdExtZvl64b == RISCV::FeatureStdExtZvl32b + 1);
+    static_assert(RISCV::FeatureStdExtZvl128b ==
+                  RISCV::FeatureStdExtZvl32b + 2);
+    static_assert(RISCV::FeatureStdExtZvl256b ==
+                  RISCV::FeatureStdExtZvl32b + 3);
+    static_assert(RISCV::FeatureStdExtZvl512b ==
+                  RISCV::FeatureStdExtZvl32b + 4);
+    static_assert(RISCV::FeatureStdExtZvl1024b ==
+                  RISCV::FeatureStdExtZvl32b + 5);
+    static_assert(RISCV::FeatureStdExtZvl2048b ==
+                  RISCV::FeatureStdExtZvl32b + 6);
+    static_assert(RISCV::FeatureStdExtZvl4096b ==
+                  RISCV::FeatureStdExtZvl32b + 7);
+    static_assert(RISCV::FeatureStdExtZvl8192b ==
+                  RISCV::FeatureStdExtZvl32b + 8);
+    static_assert(RISCV::FeatureStdExtZvl16384b ==
+                  RISCV::FeatureStdExtZvl32b + 9);
+    static_assert(RISCV::FeatureStdExtZvl32768b ==
+                  RISCV::FeatureStdExtZvl32b + 10);
+    static_assert(RISCV::FeatureStdExtZvl65536b ==
+                  RISCV::FeatureStdExtZvl32b + 11);
```

Comment:
I agree the query on Zve64x should use `hasFeature`, but I feel like the effort required to use `hasFeature` for Zvl here is a little too much...maybe the code will be shorter if we explicitly enumerate all `RISCV::FeatureStdExtZvlXXX`? Something like
```
const unsigned ZvlFeatures[] = {RISCV::FeatureStdExtZvl32b, RISCV::FeatureStdExtZvl64b, ...};
for (auto [Idx, Feature] = enumerate(ZvlFeatures)) {
  if (STI.hasFeature(Feature))
    ZvlVLen = std::max(ZvlVLen, std::pow(2, Idx + 5));
}
```

Reply:
i considered dropping most of the static_asserts